airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -305,3 +306,43 @@ definitions:
           oauth2Specification:
             description: If the connector supports OAuth, this field should be non-null.
             "$ref": "#/definitions/OAuth2Specification"
+      advanced_auth:
+        type: object
+        properties:
+          auth_flow_type:
+            type: string
+            enum: ["oauth2.0"] # Future auth types should be added here
+          predicate_key:
+            description: Json Path to a field in the connectorSpecification that should exist for the advanced auth to be applicable.
+            type: array
+            items:
+              type: string
+          predicate_value:
+            description: Value of the predicate_key fields for the advanced auth to be applicable.
+            type: string
+          oauth2_specification:
+            "$ref": "#/definitions/OAuth2ConfigSpecification"
+  OAuth2ConfigSpecification:
+    type: object
+    properties:
+      input_specification:
+        description: "OAuth2 specific blob.
+          Must be a valid non-nested JSON that refers to properties from ConnectorSpecification.connectionSpecification
+          using special annotation 'path_in_connector_config'."
+        type: object
+        existingJavaType: com.fasterxml.jackson.databind.JsonNode
+      parameter_specification:
+        description: OAuth2 specific blob. Must be a valid non-nested JSON to be injected at runtime for using OAuth flow APIs.
+        type: object
+        existingJavaType: com.fasterxml.jackson.databind.JsonNode
+      connector_auth_root_path:
```

Comment:
is there a reason anymore to use the `rootPath` convention if we already have `path_in_connection_config` param?

Reply:
- WIth `path_in_connection_config` you have to specify it for each and every property (provides flexibility). 
- Using `rootPath` introduces a "global" setting, so all outputs should end up under the same `rootPath` and don't need to be repeated (provides compactness, DRY).

Having both options makes it more complete and convenient?

Reply:
These are used for outputs:
- `path_in_connection_config` provides the **flexibility** to output parameters and overwrite values that may be spread out in the connector config under different nesting nodes. However it needs to be repeated and specified for each and every property.
- Having a "global" `rootPath` would provide the **convenience** of setting the "default" output location once and keeping it DRY. I would expect most of the cases wanting to output to the same root node location in the connector config most of the time.